src/core/lib/math/nbtheory2.cpp

```diff
@@ -247,29 +244,19 @@ uint32_t FindAutomorphismIndex2nComplex(int32_t i, uint32_t m) {
     if (i == 0) {
         return 1;
     }
-
-    // conjugation automorphism
-    if (i == int32_t(m - 1)) {
-        return uint32_t(i);
+    else if (i == (static_cast<int32_t>(m) - 1)) { // could be true if (i > 0)
+        return static_cast<uint32_t>(i);
     }
-    else {
-        // generator
-        int32_t g0;
-
-        if (i < 0) {
-            g0 = NativeInteger(5).ModInverse(m).ConvertToInt();
-        }
-        else {
-            g0 = 5;
-        }
-        uint32_t i_unsigned = (uint32_t)std::abs(i);
 
-        int32_t g = g0;
-        for (size_t j = 1; j < i_unsigned; j++) {
-            g = (g * g0) % m;
-        }
-        return uint32_t(g);
+    // conjugation automorphism
+    // generator. the usage of uint64_t prevents occasional integer overflow if the result of (g*g0) is too high
+    const uint64_t g0   = (i < 0) ? NativeInteger(5).ModInverse(m).ConvertToInt() : 5;
+    uint64_t g          = g0;
+    uint32_t i_unsigned = static_cast<uint32_t>(std::abs(i));
+    for (size_t j = 1; j < i_unsigned; j++) {
+        g = (g * g0) % m;
```

Comment:
m is always a power of two. So we can use bit shifting instead of %. Just in case, check that m is a power of two at the beginning.